### PR TITLE
Group account creation

### PR DESCRIPTION
**Pull request by aaronc**
Adds a hack to allow us to use auth.AccountKeeper in the group module with ADR 033 and uses this to create group module accounts.

### CODE REVIEW

**Comment by aaronc** [comment]
There are some failing tests here that I'm not too sure about. Can you check @blushi ? Looks like otherwise things are mostly wired up.

File: app/app.go [hunk 1]
```diff
@@ -359,10 +359,19 @@ func NewRegenApp(logger log.Logger, db dbm.DB, traceStore io.Writer, loadLatest
...
 
-	err := newModuleManager.RegisterModules(NewModules)
+	// BEGIN HACK: this is a total, ugly hack until x/auth supports ADR 033 or we have a suitable alternative
+	groupModule := group.Module{AccountKeeper: authkeeper.AccountKeeper{}}
+	// use a separate newModules from the global NewModules here because we need to pass state into the group module
+	newModules := []moduletypes.Module{
+		ecocredit.Module{},
+		data.Module{},
+		groupModule,
+	}
```
Comment (robert-zaremba) [inline app/app.go:369]: In line 137 we have:
```
	NewModules = []moduletypes.Module{
		ecocredit.Module{},
		data.Module{},
		group.Module{},
	}
```.
It seams we should remove that.

File: app/app.go [hunk 2]
```diff
@@ -359,10 +359,19 @@ func NewRegenApp(logger log.Logger, db dbm.DB, traceStore io.Writer, loadLatest
 	/* New Module Wiring START */
 	newModuleManager := servermodule.NewManager(app.BaseApp, codec.NewProtoCodec(interfaceRegistry))
 
-	err := newModuleManager.RegisterModules(NewModules)
+	// BEGIN HACK: this is a total, ugly hack until x/auth supports ADR 033 or we have a suitable alternative
```
Comment (robert-zaremba) [inline app/app.go:362]: Is this a hack or a TODO? 

Why we can't use a `MsgServer` client here?

File: types/module/server/module_key.go [hunk 3]
```diff
@@ -1,14 +1,15 @@
...
+	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/regen-network/regen-ledger/types"
 )
 
 type ModuleKey interface {
 	types.InvokerConn
 
 	ModuleID() types.ModuleID
-	Address() []byte
+	Address() sdk.AccAddress
```
Comment (robert-zaremba) [inline types/module/server/module_key.go:12]: :+1:

File: x/group/conditions.go [hunk 4]
```diff
@@ -1,130 +0,0 @@
-package group
```
Comment (robert-zaremba) [inline x/group/conditions.go:1]: why removing this file?

Comment (aaronc) [reply x/group/conditions.go:1]: It's replaced by ADR 028/033

File: app/app.go (hunk 2 above)
Comment (aaronc) [reply app/app.go:362]: It's a brutal hack

File: app/app.go (hunk 1 above)
Comment (aaronc) [reply app/app.go:369]: Because cli code still uses it. This is strictly a hack until x/auth supports ADR 033

File: app/app.go (hunk 2 above)
Comment (aaronc) [reply app/app.go:362]: x/auth has no MsgServer

**Comment by aaronc** [comment]
Found a slightly cleaner way I think, but still a bunch of test failures...

**Comment by blushi** [comment]
> Found a slightly cleaner way I think, but still a bunch of test failures...

@aaronc I've fixed the tests as part of https://github.com/regen-network/regen-ledger/pull/252

**Comment by aaronc** [comment]
Thanks @blushi !

**Comment by aaronc** [comment]
What do you think otherwise @blushi ?

**Review by blushi — APPROVED** [review]
Pre-approving, hacky but does the job for now.

File: app/app.go (hunk 1 above)
Comment (aaronc) [reply app/app.go:369]: Because they are used for setting up cli commands

**Comment by aaronc** [comment]
Super hacky. Please let's try to think of better solutions 😉

**Pull request merged**